packages/core/src/plugins/use-masked-errors.ts

```diff
@@ -0,0 +1,87 @@
+import { Plugin, ExecutionResult } from '@envelop/types';
+import { handleStreamOrSingleExecutionResult } from '../utils.js';
+
+export const DEFAULT_ERROR_MESSAGE = 'Unexpected error.';
+
+export type MaskErrorFn = (error: unknown, message: string) => Error;
+
+export type SerializableGraphQLErrorLike = Error & {
+  name: 'GraphQLError';
+  toJSON(): { message: string };
+};
+
+export function isGraphQLError(error: unknown): error is Error & { originalError?: Error } {
+  return error instanceof Error && error.name === 'GraphQLError';
+}
+
+export function createSerializableGraphQLError(message: string): SerializableGraphQLErrorLike {
+  const error = new Error(message);
+  error.name = 'GraphQLError';
+  Object.defineProperty(error, 'toJSON', {
+    value() {
+      return {
+        message: error.message,
+      };
+    },
+  });
+  return error as SerializableGraphQLErrorLike;
+}
+
+export const defaultMaskErrorFn: MaskErrorFn = (err, message) => {
+  if (isGraphQLError(err)) {
+    if (err?.originalError) {
+      if (isGraphQLError(err.originalError)) {
+        return err;
+      }
+      return createSerializableGraphQLError(message);
+    }
+    return err;
+  }
+  return createSerializableGraphQLError(message);
+};
+
+export type UseMaskedErrorsOpts = {
+  /** The function used for identify and mask errors. */
+  maskErrorFn?: MaskErrorFn;
```

Comment:
I personally don't like the `Fn` appendix, Could we name this just MaskError and maskError?